### PR TITLE
Fix MISRA violation

### PR DESCRIPTION
**Pull request by AniruddhaKanhere**
*Issue #, if available:*

*Description of changes:*
This PR fixes one MISRA violation;
Updates the coverity config file
Adds conditional compilation option to test CMake

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tools/coverity/misra.config [hunk 1]
```diff
@@ -1,27 +1,25 @@
...
+            "deviation": "Rule 2.5",
+            "reason": "Allow unused macros. Library headers may define macros intended for the application's use, but not used by a specific file."
         },
         {
-            deviation: "Rule 3.1",
-            category: "Required",
-            reason: "Allow nested comments. Documentation blocks contain comments for example code."
+            "deviation": "Rule 3.1",
+            "category": "Required",
+            "reason": "Allow nested comments. Doßcumentation blocks contain comments for example code."
```
Comment (moninom1) [inline tools/coverity/misra.config:18]: Doßcumentation - Is this a typo?

**Comment by Skptak** [comment]
Will this repo need the same change to the CI build as [coreMQTT's ci.yml file](https://github.com/FreeRTOS/coreMQTT/pull/276/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f)?

Will it also need its unit test version bumped up as was done in [coreMQTT/test/CMakeList.txt](https://github.com/FreeRTOS/coreMQTT/pull/276/files#r1494852586)?

File: tools/coverity/misra.config (hunk 1 above)
Comment (AniruddhaKanhere) [reply tools/coverity/misra.config]: Indeed it is. Nice catch. Fixed.

**Comment by AniruddhaKanhere** [comment]
> Will this repo need the same change to the CI build as [coreMQTT's ci.yml file](https://github.com/FreeRTOS/coreMQTT/pull/276/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03f)?
> 
> Will it also need its unit test version bumped up as was done in [coreMQTT/test/CMakeList.txt](https://github.com/FreeRTOS/coreMQTT/pull/276/files#r1494852586)?

Fixed!

**Review by moninom1 — APPROVED** [review]

**Review by Skptak — APPROVED** [review]

**Review by paulbartell — APPROVED** [review]

**Pull request merged**